libs/jsonforms-components/src/lib/Controls/FormStepper/FormStepper.spec.tsx

```diff
@@ -0,0 +1,52 @@
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/jest-globals';
+import '@testing-library/jest-dom';
+import FormStepper, { getFormFieldValue, renderFormFields, resolveLabelFromScope } from './FormStepperControl';
+import {
+  Categorization,
+  UISchemaElement,
+  deriveLabelForUISchemaElement,
+  Category,
+  StatePropsOfLayout,
+  isVisible,
+  isEnabled,
+} from '@jsonforms/core';
+import Ajv from 'ajv';
+import { AjvProps, withAjvProps } from '@jsonforms/material-renderers';
+import { TranslateProps, withJsonFormsLayoutProps, withTranslateProps, useJsonForms } from '@jsonforms/react';
+import { mock } from 'node:test';
+export interface CategorizationStepperLayoutRendererProps extends StatePropsOfLayout, AjvProps, TranslateProps {
+  // eslint-disable-next-line
+  data: any;
+}
+
+//mock data
+const data = {
+  firstName: 'Alex',
+  address: {
+    street: 'Springfield',
+  },
+};
+
+const translator = {
+  id: '',
+  defaultMessage: '',
+  values: '',
+};
+
+describe('resolveLabelFromScope', () => {
+  it('Should correctly resolve and format label from scope', () => {
+    const scope1 = '#/properties/firstName';
+    const scope2 = '#/properties/address/properties/street';
+    expect(resolveLabelFromScope(scope1)).toEqual('First Name');
+    expect(resolveLabelFromScope(scope2)).toEqual('Street');
+  });
+});
+describe('getFormFieldValue', () => {
+  it('Should return the correct value for a given scope', () => {
+    const scope1 = '#/properties/firstName';
+    const scope2 = '#/properties/address/properties/street';
+    expect(getFormFieldValue(scope1, data)).toEqual('Alex');
+    expect(getFormFieldValue(scope2, data)).toEqual('Springfield');
+  });
+});
```

Comment:
This is your opportunity to test for bad data as well.  Like, what happens if the scope is malformed e.g. scope="properties.address.properties.street"?  Or, what happens if properties is misspelled? Or, what if scope is null or an empty string?  Unit testing is as much how the code is handling edge cases as it is about good data.  Try to adopt the mindset of a tester - I'm going to do whatever I can to make this code fail!

btw, I could be wrong, but shouldn't the label be 'First name'?  I thought we were using sentence case?